easybuild/easyconfigs/r/Relate/Relate-20250429-foss-2023b.eb

```diff
@@ -0,0 +1,79 @@
+# Author: Jasper Grimm (UoY)
+easyblock = 'CMakeMake'
+
+name = 'Relate'
+version = '20250429'
+local_commit = '7bd53c7'
+
+homepage = 'https://myersgroup.github.io/relate/'
+description = "Software for estimating genome-wide genealogies for thousands of samples"
+citing = """
+[1] Leo Speidel, Marie Forest, Sinan Shi, Simon Myers. A method for estimating genome-wide
+    genealogies for thousands of samples. Nature Genetics 51: 1321-1329, 2019.
+[2] Leo Speidel, Lara Cassidy, Robert W. Davies, Garrett Hellenthal, Pontus Skoglund, Simon Myers.
+    Inferring population histories for ancient genomes using genome-wide genealogies.
+    bioRxiv:2021.02.17.431573.
+"""
+
+toolchain = {'name': 'foss', 'version': '2023b'}
+
+github_account = 'MyersGroup'
+source_urls = [GITHUB_LOWER_SOURCE]
+sources = ['%s.tar.gz' % local_commit]
+patches = ['%(name)s-20211123_add-cmake-install-targets.patch']
+checksums = [
+    {'7bd53c7.tar.gz': 'eb0c48cca3f053e0b51ca86d5df3c806b478d96a88ee570b438491447392e497'},
+    {'Relate-20211123_add-cmake-install-targets.patch':
+     'd4565aeef0036013b82b37e00909965a6447ab14b154caa3e304307d6f7818e4'},
+]
+
+builddependencies = [('CMake', '3.27.6')]
+
+dependencies = [
+    # Python (numpy), and R required for some scripts
+    ('Python', '3.11.5'),
+    ('SciPy-bundle', '2023.11'),
+    ('R', '4.4.1'),
+]
+
+runtest = 'test'
+
+# script directories to add to path:
+local_script_dirs = ['DetectSelection', 'EstimatePopulationSize', 'PrepareInputFiles', 'RelateParallel',
+                     'SampleBranchLengths', 'TreeView']
+# un/comment as appropriate:
+# local_script_dirs += ['RelateLSF']
+# local_script_dirs += ['EstimatePopulationSizeSGE', 'RelateSGE']
+# local_script_dirs += ['RelateSlurm']
+
+local_bins = ['bin/%s%s' % (name, x) for x in ['', 'CoalescentRate', 'Extract', 'FileFormats',
+                                               'MutationRate', 'Selection', 'TreeView']]
+local_libs = ['lib/libtskitStatic.a'] + [
+    'lib/lib%s%s' % (x, y) for x in ['gzstream', 'kastore', 'relate'] for y in ['Shared.%s' % SHLIB_EXT, 'Static.a']
+]
+
+sanity_check_paths = {
+    'files': local_bins + local_libs,
+    'dirs': ['bin', 'lib', 'example'] + ['scripts/%s' % x for x in local_script_dirs],
+}
+
+sanity_check_commands = [
+    "%(name)s --help",
+    "cd %(builddir)s/relate-*/example && gunzip data/*.gz && Relate --mode All -m 1.25e-8 -N 30000 \
```

Comment:
This does not work with `--module-only`, as the build dir is no longer available.

Reply:
If this not a bit much for sanity checking? It should not be used as a replacement for tests.

```
test_cmd = ...
```
should work?

Reply:
This kinda stuff is what the `test_cases_step` is for, I think?

Reply:
Done in https://github.com/easybuilders/easybuild-easyconfigs/pull/22859/commits/d91d3abad256b30ed1a30a0bab1596d521d6d093. I also tried to use `tests`, but that expects a script, and that didn't really work here. So I've added it to `test_cmd` instead.